chalicelib/lib/slack.py

```diff
@@ -25,13 +25,31 @@ def post_message(self, message: str, channel: str) -> requests.models.Response:
         :param channel: The channel to send the message
         :return: requests.models.Response
         """
-        return requests.post(
-            url=f"{self.slack_api_url}/chat.postMessage",
-            json={"channel": channel, "text": f"{message}"},
-            headers=self.headers,
+        return self._handle_response(
+            requests.post(
+                url=f"{self.slack_api_url}/chat.postMessage",
+                json={"channel": channel, "text": f"{message}"},
+                headers=self.headers,
+            )
         )
 
 
+    def _handle_response(self, response: requests.models.Response) -> None:
+        """
+        Check the response from slack.
+        A normal slack response should always be JSON with a key "ok" with value True for the response to be valid
+
+        """
+        try:
+            validated_response = response.json()
+            if not validated_response.get('ok'):
+                log.critical(f"Slack responded with not ok. Message was: {validated_response}")
+        except AttributeError as error:
+            log.critical(f"Unable get valid json from response. Error was: {error}", exc_info=True)
+        
+        return None
```

Comment:
Just realised that maybe it should return the `validated_response` object so instances of this class can inspect the result